kong/api/routes/upstreams.lua

```diff
@@ -107,4 +109,54 @@ return {
       crud.post(self.params, dao_factory.targets)
     end,
   },
+
+  ["/upstreams/:name_or_id/targets/active/"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
+
+    GET = function(self, dao_factory)
+      self.params.active = nil
+
+      local target_history, err = dao_factory.targets:find_all({
+        upstream_id = self.params.upstream_id,
+      })
+
+      if not target_history then
+        return app_helpers.yield_error(err)
+      end
+
+      --sort and walk based on target and creation time
+      for _, target in ipairs(target_history) do
+        target.order = target.target..":"..target.created_at..":"..target.id
+      end
+      table.sort(target_history, function(a, b) return a.order > b.order end)
+
+      local ignored = {}
+      local found = {}
+      local found_n = 0
```

Comment:
not sure if this is needed in such cold code paths, it just adds cognitive load where one could use `table.insert` instead imho. But it's fine.